File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-7.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findMin(nums) {
        let left = 0
        let right = nums.length - 1
        let curmin = nums[0]

        while (left <= right){
            if (nums[left] < nums[right]){
                curmin = Math.min(curmin, nums[left])
                return curmin
            } //this means the smallest value could be the leftmost

            let middle = Math.trunc(left + ((right - left) / 2))
            curmin = Math.min(curmin, nums[middle])

            if (nums[middle] > nums[right]){
                left = middle + 1
            }
            else{
                right = middle - 1
            }
        }

        return curmin
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-7.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxP = 0
        left, right = 0, 1

        while right < len(prices):
            if prices[right] < prices[left]:
                left = right
                right += 1
                continue
            else:
                print(prices[right] - prices[left])
                maxP = max(maxP, (prices[right] - prices[left]))
                right += 1
        
        return maxP


        
        